plutus-core/plutus-core/src/PlutusCore/Default/Builtins.hs

```diff
@@ -603,7 +603,7 @@ argument as a 'Bool', we could do that:
       where
         idAssumeCheckBoolPlc :: Opaque val Bool -> EvaluationResult Bool
         idAssumeCheckBoolPlc val =
-            case asConstant @_ @UnliftingError Nothing val of
+            case asConstant val of
```

Comment:
Is it possible to have a builtin function whose plutus type is `bool -> bool`, which returns `EvaluationFailure` if the argument type is not `bool`, but also doesn't require the argument to be a constant? It seems none of `IdBool`, `IdAssumeBool` or `IdAssumeCheckBool` does that.

Reply:
I think that's impossible: the only *values* of type `bool` are constants.

Reply:
> I think that's impossible: the only _values_ of type `bool` are constants.

The only `CekValue`s. We could have a lazy evaluator and pass `error` as an argument. The builtins machinery should be able to handle that and pass `error` through and the docs are about builtins.